lib/puppet/util/servicenow.rb

```diff
@@ -89,4 +89,54 @@ def do_snow_request(uri, http_verb, body, user: nil, password: nil, oauth_token:
     end
   end
   module_function :do_snow_request
+
+  def requested_status?(status, corrective_change, noop_pending)
+    change_types = settings['incident_creation_report_attributes']
+
+    if change_types.include? 'none'
+      Puppet.info('servicenow reporting: incident_creation_report_attributes includes \'none\'.')
+      return false
+    elsif change_types.include? 'unchanged'
+      Puppet.info('servicenow reporting: incident_creation_report_attributes includes \'unchanged\'.')
+      return true
+    elsif change_types.include? 'all'
+      change_types = ['failed_changes', 'corrective_changes', 'intentional_changes', 'pending_changes']
+    end
+
+    report_on_noop = change_types.include? 'pending_changes'
+
+    if status == 'failed' && change_types.include?('failed_changes')
+      Puppet.info('servicenow reporting: decision: reportable failure')
+      return true
+    end
+
+    if noop_pending && !report_on_noop
+      Puppet.info('servicenow reporting: pending change reporting disabled')
+      return false
+    elsif noop_pending && report_on_noop
+      # This branch is a problem. Currently, given the data available to report
+      # processors like ours, on noop runs (pending changes), only corrective
```

Comment:
Not sure what this means. Is it basically `pending_correctional_change`/`pending_intentional_change` ?

Reply:
This whole block has been shifted around at this point, but yes that was the original intent, the inability to distinguish between the two kinds of pending changes.

Reply:
Ah I just reread this comment. It's worded exactly how we are talking about this in Slack DM. Cool. thanks.